crates/ruma-common/src/events/room/message/sanitize.rs

```diff
@@ -0,0 +1,210 @@
+#[cfg(feature = "sanitize")]
+pub fn remove_html_reply_fallback(s: &str) -> String {
+    let sanitizer = HtmlSanitizer::reply_fallback_remover();
+    sanitizer.clean(s)
+}
+
+/// Remove the [rich reply fallback] of the given plain text string.
+///
+/// [rich reply fallback]: https://spec.matrix.org/v1.2/client-server-api/#fallbacks-for-rich-replies
+pub fn remove_plain_reply_fallback(s: &str) -> &str {
+    if !s.starts_with("> ") {
+        s
+    } else {
+        let mut start = 0;
+        for (pos, _) in s.match_indices('\n') {
+            if !&s[pos + 1..].starts_with("> ") {
+                start = pos + 1;
+                break;
+            }
+        }
+        &s[start..]
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::remove_plain_reply_fallback;
+    #[cfg(feature = "sanitize")]
+    use super::{
+        remove_html_reply_fallback, sanitize_html, HtmlSanitizerMode, RemoveReplyFallback,
+    };
+
+    #[test]
+    #[cfg(feature = "sanitize")]
+    fn sanitize() {
+        let sanitized = sanitize_html(
+            "\
+            <mx-reply>\
+                <blockquote>\
+                    <a href=\"https://matrix.to/#/!n8f893n9:example.com/$1598361704261elfgc:localhost\">In reply to</a> \
+                    <a href=\"https://matrix.to/#/@alice:example.com\">@alice:example.com</a>\
+                    <br>\
+                    Previous message\
+                </blockquote>\
+            </mx-reply>\
+            <removed>This has no tag</removed>\
+            <p>But this is inside a tag</p>\
+            ",
+            HtmlSanitizerMode::Strict,
+            RemoveReplyFallback::No,
+        );
+
+        assert_eq!(
+            sanitized,
+            "\
+            <mx-reply>\
+                <blockquote>\
+                    <a href=\"https://matrix.to/#/!n8f893n9:example.com/$1598361704261elfgc:localhost\">In reply to</a> \
+                    <a href=\"https://matrix.to/#/@alice:example.com\">@alice:example.com</a>\
+                    <br>\
+                    Previous message\
+                </blockquote>\
+            </mx-reply>\
+            This has no tag\
+            <p>But this is inside a tag</p>\
+            "
+        );
+    }
+
+    #[test]
+    #[cfg(feature = "sanitize")]
+    fn sanitize_without_reply() {
+        let sanitized = sanitize_html(
+            "\
+            <mx-reply>\
+                <blockquote>\
+                    <a href=\"https://matrix.to/#/!n8f893n9:example.com/$1598361704261elfgc:localhost\">In reply to</a> \
+                    <a href=\"https://matrix.to/#/@alice:example.com\">@alice:example.com</a>\
+                    <br>\
+                    Previous message\
+                </blockquote>\
+            </mx-reply>\
+            <removed>This has no tag</removed>\
+            <p>But this is inside a tag</p>\
+            ",
+            HtmlSanitizerMode::Strict,
+            RemoveReplyFallback::Yes
```

Comment:
I wonder why rustfmt isn't doing this automatically..

```suggestion
            RemoveReplyFallback::Yes,
```